### ISSUE TITLE
Larger Badges for Extension Installer UI

### ISSUE THREAD

**Issue by tgeorgeux** [jupyterlab/jupyterlab#8043]
@marsbarlee and I have been working on showing larger badges so people can more easily identify the source of their extensions. 

## Why is this important?
There are various extensions, some of which are similar, from different sources. Some people may only want to download extensions released by Jupyter, or another org. In addition, it incentivizes others to build and maintain extensions (by giving recognition through the UI). 

## How should we implement?
Here is a prototype of the badges @marbarslee already posted in #7967 .
![image](https://user-images.githubusercontent.com/22109376/76687618-c330f000-65e2-11ea-968f-15ea54f32b2b.png)

We played around with a couple of sizes and 32x32 with an 8px margin around seems good.

**Comment by krassowski** [comment]
Great work, I am so happy to see it going forward!

Responding to the question from the other issue:
> For extensions from Github users, we will be using their Github icons or identicons. What do others think?

I am not sure if I like my profile picture as much as to see it in the extension panel ;) However, it would be great if this would be opt-in!

Potentially related - what would there be a distinction between icons for https://github.com/jupyterlab and https://github.com/jupyter-widgets extensions and others? Let's say I create jupyterlab-lsp org but decide not to follow the due process (I won't), how would users know that it is not as legit as the others? Are those "developed by Jupyter" indicators (below) going away with the introduction of white/black lists? I liked them a lot!

![developed_by_jupyter](https://user-images.githubusercontent.com/5832902/76707670-10978500-66e9-11ea-8658-f6ccb2629b0e.png)

Also, not sure if having ten 32x32 jupyter icons would be informative (most of the very good extensions are from JupyterLab org as for today).

**Comment by echarles** [comment]
> Are those "developed by Jupyter" indicators (below) going away with the introduction of white/black lists? I liked them a lot!

The work done in the black/white listings does not remove the  "developed by Jupyter" icons. Strangely, the code was using a method `isWhitelisted` to create those icons. I have renamed that method to `isJupyterOrg` to avoid confusion. This only impact code and it was a private method anyway. On the user side, b/w listing does not change anything for the icons.

**Comment by echarles** [comment]
> For extensions from Github users, we will be using their Github icons or identicons. What do others think?

GitHub has a trick for this https://github.com/jupyterlab.png?size=200 - Sounds like super easy.

**Comment by echarles** [comment]
https://github.com/jupyterlab.png?size=32 for 32px size

**Comment by echarles** [comment]
The NPM response contains details on the github repository in the `homepage` attributed. This can be used to extract the repo organisation/account and generate the icons. Please not that some extensions do not state `homepage` or state a homepage being not github. What to show in those cases? (default icon, no icon...)?

```json
        "links": {
          "npm": "https://www.npmjs.com/package/jupyterlabtd_git",
          "homepage": "https://github.com/jupyterlab/jupyterlab-git",
          "repository": "https://github.com/jupyterlab/jupyterlab-git",
          "bugs": "https://github.com/jupyterlab/jupyterlab-git/issues"
        },
```

**Comment by vidartf** [comment]
I don't think we should hit github for icons. We might already need to put the extension manager behind a privacy notice (linking to the privacy policy of npm), and I wouldn't want to add further information leaks.

**Comment by tgeorgeux** [comment]
That makes sense. Should we add a neat little spec that the extension installer can call out to display a badge? It makes sense to me that extension developers may want to identify themselves.

**Comment by vidartf** [comment]
> Should we add a neat little spec that the extension installer can call out to display a badge?

If we can think of one that won't generate a lot of extra network requests, sure!

**Comment by echarles** [comment]
Regarding the privacy notice, this may help `npm, Inc., is being purchased by GitHub.` https://github.blog/2020-03-16-npm-is-joining-github/

For the HTTP Requests, I guess the current pagination can mitigate this. Also the icons would have to be displayed after all the other resources loaded so would not impact the jlab load time.

**Comment by jasongrout** [comment]
> Regarding the privacy notice, this may help `npm, Inc., is being purchased by GitHub.` https://github.blog/2020-03-16-npm-is-joining-github/

Oh, wow, thanks for bringing that to our attention!

**Comment by echarles** [comment]
@jasongrout wow indeed. Who will be next?

**Comment by telamonian** [comment]
This was resolved by #8078
